solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java

```diff
@@ -70,20 +67,13 @@ public void doGet(HttpServletRequest _request, HttpServletResponse _response) th
         // We have to close this to flush OutputStreamWriter buffer
         out =
             new OutputStreamWriter(
-                new CloseShieldOutputStream(response.getOutputStream()), StandardCharsets.UTF_8);
+                CloseShieldOutputStream.wrap(response.getOutputStream()), StandardCharsets.UTF_8);
 
-        String html = IOUtils.toString(in, "UTF-8");
         Package pack = SolrCore.class.getPackage();
-
-        String[] search = new String[] {"${contextPath}", "${adminPath}", "${version}"};
-        String[] replace =
-            new String[] {
-              StringEscapeUtils.escapeEcmaScript(request.getContextPath()),
-              StringEscapeUtils.escapeEcmaScript(CommonParams.CORES_HANDLER_PATH),
-              StringEscapeUtils.escapeEcmaScript(pack.getSpecificationVersion())
-            };
-
-        out.write(StringUtils.replaceEach(html, search, replace));
+        String html =
+            IOUtils.toString(in, StandardCharsets.UTF_8)
+                .replace("${version}", pack.getSpecificationVersion());
+        out.write(html);
```

Comment:
Guessing we don't need the other replacements?

Reply:
Very thorough of you -- much appreciated!